src/utils/redirect.ts

```diff
@@ -5,7 +5,13 @@ import { UserType } from '@/types/interfaces'
 
 export const redirectIfTaskCta = (searchParams: Record<string, string>) => {
   const taskId = z.string().safeParse(searchParams.taskId)
+  const commentId = z.string().safeParse(searchParams.commentId)
   if (taskId.data) {
+    if (commentId) {
+      redirect(
+        `${apiUrl}/detail/${taskId.data}/iu?token=${z.string().parse(searchParams.token)}&commentId=${commentId}&isRedirect=1`,
```

Comment:
This only redirects to IU detail, not Client detail. So on clicking the email button the client will be redirected to IU task detail view.

I think the way to go here is to check tokenPayload for `internalUserId` or `clientId`, and handle redirection accordingly